iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractTerminal.java

```diff
@@ -70,11 +73,14 @@ public void setNum(int num) {
 
     @Override
     public double getP() {
-        return p.get(network.get().getVariantIndex());
+        return Optional.ofNullable(network).map(n -> p.get(n.get().getVariantIndex())).orElse(Double.NaN);
     }
 
     @Override
     public Terminal setP(double p) {
+        if (network == null) {
+            throw new PowsyblException("Can not modify removed equipment");
```

Comment:
An even smaller detail, we write "cannot" in the other exceptions, but both are ok anyway